spec/aws/active_job/sqs/job_runner_spec.rb

```diff
@@ -4,14 +4,121 @@ module Aws
   module ActiveJob
     module SQS
       describe JobRunner do
+        subject { described_class.new(msg) }
+
         let(:job_data) { TestJob.new('a1', 'a2').serialize }
+        let(:event_body) { 'example sqs message' }
         let(:body) { ActiveSupport::JSON.dump(job_data) }
-        # message is a reserved minitest name
-        let(:msg) { double(data: double(body: body)) }
+        let(:message_id) { SecureRandom.uuid }
+        let(:active_job_attributes) do
+          {
+            'aws_sqs_active_job_class' => { 
+              'string_value' => 'TestJob',
+              'data_type' => 'String'
+            },
+            'aws_sqs_active_job_version' => {
+              'string_value' => '1.0.0',
+              'data_type' => 'String'
+            }
+          }
+        end
+        let(:msg) do # message is a reserved minitest name
+          double( data: double(body: body),
+                  message_id: message_id,
+                  queue_url: queue_config.dig(:default_queue, :url),
+                  message_attributes: active_job_attributes,
+                  receipt_handle: SecureRandom.uuid)
+        end
+        let(:event_msg) do
+          double( data: double(message_id: SecureRandom.uuid, body: event_body, message_attributes: {}),
+                  queue_url: queue_config.dig(:event_queue, :url),
+                  message_id: message_id,
+                  message_attributes: {},
+                  receipt_handle: SecureRandom.uuid)
+        end
+        let(:queue_config) do
+          {
+            default_queue: {
+              url: 'http://example.sqs/default_queue',
+            },
+            event_queue: {
+              url: 'http://example.sqs/event_queue',
+              event_message_class: 'EventJob'
+            }
+          }
+        end
 
-        it 'parses the job data' do
-          job_runner = JobRunner.new(msg)
-          expect(job_runner.instance_variable_get(:@job_data)).to eq job_data
+        before do
+          described_class.class_variable_set(:@@queue_handlers, nil)
```

Comment:
I don't like this - more reason that it belongs in Configuration, and then a test instance of Configuration can automatically return these values (not really a detail to test here).